apps/gitness/src/pages/project-settings/project-settings-members-page.tsx

```diff
@@ -0,0 +1,181 @@
+import { timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
+import { useQueryState, parseAsInteger } from 'nuqs'
+import { PageResponseHeader } from '../../types'
+
+const filterOptions = [{ name: 'Filter option 1' }, { name: 'Filter option 2' }, { name: 'Filter option 3' }]
+const sortOptions = [{ name: 'Sort option 1' }, { name: 'Sort option 2' }, { name: 'Sort option 3' }]
+
+const ProjectSettingsMemebersPage = () => {
+  const space_ref = useGetSpaceURLParam()
+  const [totalMembers, setTotalMembers] = useState<number | null>(null)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+  const [dialogState, setDialogState] = useState({
+    isDialogEditOpen: false,
+    isDialogDeleteOpen: false,
+    selectedMember: null as { display_name: string; role: string; email: string } | null
+  })
+
+  const { sort, query: currentQuery } = useCommonFilter<MembershipListQueryQueryParams['sort']>()
+  const [query, _] = useQueryState('query', { defaultValue: currentQuery || '' })
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+
+  const { isLoading, data: { body: members, headers } = {} } = useMembershipListQuery(
+    { space_ref: space_ref ?? '', queryParams: { query, sort } },
+    {
+      onSuccess: ({ body: members }) => {
+        setTotalMembers(members.length) // Update total members count
+        console.log(members.length)
+      }
+    }
+  )
+
+  const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
+
+  const handleDelete = () => {
+    setIsDeleting(true)
+    setTimeout(() => {
+      setIsDeleting(false)
+      setDialogState(prev => ({ ...prev, isDialogDeleteOpen: false, selectedMember: null }))
+    }, 2000)
+  }
+
+  const handleRoleSave = () => {
+    setIsSubmitting(true)
+    setTimeout(() => {
+      setIsSubmitting(false)
+      setDialogState(prev => ({ ...prev, isDialogEditOpen: false, selectedMember: null }))
+    }, 2000)
+  }
+
+  const renderMemberListContent = () => {
+    if (isLoading) return <SkeletonList />
+    if (!members?.length) {
+      return (
+        //add this layout to target the content in the center of the page without header and subheader
+        <SandboxLayout.Main hasLeftPanel>
+          <SandboxLayout.Content maxWidth="3xl" className="h-screen">
+            <NoData
+              iconName="no-data-members"
+              title="No Members yet"
+              description={['Add your first team members by inviting them to join this project.']}
+              primaryButton={{ label: 'Invite new members' }}
+            />
+          </SandboxLayout.Content>
+        </SandboxLayout.Main>
+      )
+    }
+    return (
+      <>
+        <MembersList
+          members={members.map((member: TypesMembershipUser) => ({
+            display_name: member.principal?.display_name ?? '',
+            role: member.role as EnumMembershipRole | string,
```

Comment:
Why do we need this type cast here? Isn't `member.role` already of type `EnumMembershipRole | string`?

Reply:
it will be changed in the following up fixes